lib/elixir/lib/access.ex

```diff
@@ -422,4 +422,73 @@ defmodule Access do
   defp all([], _next, gets, updates) do
     {:lists.reverse(gets), :lists.reverse(updates)}
   end
+
+  @doc ~S"""
+  Accesses the element at `index` (zero based) of a list.
+
+  ## Examples
+
+      iex> list = [%{name: "john"}, %{name: "mary"}]
+      iex> get_in(list, [Access.at(1), :name])
+      "mary"
+      iex> get_and_update_in(list, [Access.at(0), :name], fn
+      ...>   prev -> {prev, String.upcase(prev)}
+      ...> end)
+      {"john", [%{name: "JOHN"}, %{name: "mary"}]}
+      iex> pop_in(list, [Access.at(0), :name])
+      {"john", [%{}, %{name: "mary"}]}
+
+  When the index is out of bounds, `nil` is returned and the update function is never called:
+      iex> list = [%{name: "john"}, %{name: "mary"}]
+      iex> get_in(list, [Access.at(10), :name])
+      nil
+      iex> get_and_update_in(list, [Access.at(10), :name], fn
+      ...>   prev -> {prev, String.upcase(prev)}
+      ...> end)
+      {nil, [%{name: "john"}, %{name: "mary"}]}
+
+  An error is raised for negative indexes:
+
+      iex> get_in([], [Access.at(-1)])
+      ** (RuntimeError) Access.at/1 expected a non-negative index, got: -1
+
+  An error is raised if the accessed structure is not a list:
+
+      iex> get_in(%{}, [Access.at(1)])
+      ** (RuntimeError) Access.at/1 expected a list, got: %{}
+  """
+  def at(index) when index >= 0 do
+    fn(op, data, next) -> at(op, data, index, next) end
+  end
+
+  def at(index) do
```

Comment:
There is no need for this clause, in this particular case, it should be obvious in the stacktrace. The reason we add a explicit raise in the other cases is because it is often invoked inside a callback.